diesel/src/sqlite/mod.rs

```diff
@@ -13,3 +13,21 @@ pub mod query_builder;
 pub use self::backend::{Sqlite, SqliteType};
 pub use self::connection::SqliteConnection;
 pub use self::query_builder::SqliteQueryBuilder;
+
+/// Trait for the implementation of a SQLite aggregate function
+///
+/// This trait is to be used in conjunction with the `sql_function!`
+/// macro for defining a custom SQLite aggregate function. See
+/// the documentation [there](../../diesel_derives/macro.sql_function_proc.html) for details.
+pub trait SqliteAggregateFunction<Args>: Default {
+    /// The result type of the SQLite aggregate function
+    type Output;
+
+    /// The `step()` method is called once for every record of the query
+    fn step(&mut self, args: Args);
+    /// After the last row has been processed, the `finalize()` method is
+    /// called to compute the result of the aggregate function. If no rows
+    /// were processed, `finalize()` is not called and `None` is returned
+    /// instead
+    fn finalize(self) -> Self::Output;
```

Comment:
I decided to have `finalize` not return an `Option` since it made more sense to me not to return one. However, I did add an extra sentence in the documentation addressing when `None` would be returned

Reply:
:+1: